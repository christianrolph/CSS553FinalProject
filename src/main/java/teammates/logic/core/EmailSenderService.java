package teammates.logic.core;

import teammates.common.util.EmailSendingStatus;
import teammates.common.util.EmailWrapper;

/**
 * An email sender interface used by services for sending emails.
 */
public interface EmailSenderService {

    /**
     * Parses the {@code wrapper} email object to specific implementations of email object
     * used by the service.
     */
    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
    // accounts for the many different Exceptions from different email services
    Object parseToEmail(EmailWrapper wrapper) throws Exception;

    /**
     * Sends the email packaged as a {@code wrapper}.
     */
    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
    // accounts for the many different Exceptions from different email services
    EmailSendingStatus sendEmail(EmailWrapper wrapper) throws Exception;

}
