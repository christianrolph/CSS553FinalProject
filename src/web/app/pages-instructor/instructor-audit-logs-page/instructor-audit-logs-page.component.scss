.bg-form {
  background-color: #EAEFF5;
}

.no-border {
  border: 0;
}

.alert-success-border {
  border-color: #C3E6CB;
}

.no-activity-body {
  padding: .5rem 1rem;
}
