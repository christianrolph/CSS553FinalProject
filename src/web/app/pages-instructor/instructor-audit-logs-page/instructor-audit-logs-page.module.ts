import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule, Routes } from '@angular/router';
import { NgbDatepickerModule, NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { PanelChevronModule } from '../../components/panel-chevron/panel-chevron.module';
import { SessionEditFormModule } from '../../components/session-edit-form/session-edit-form.module';
import { SortableTableModule } from '../../components/sortable-table/sortable-table.module';
import { InstructorAuditLogsPageComponent } from './instructor-audit-logs-page.component';

const routes: Routes = [
  {
    path: '',
    component: InstructorAuditLogsPageComponent,
  },
];

/**
 * Module for instructor audit logs page
 */
@NgModule({
  declarations: [InstructorAuditLogsPageComponent],
  exports: [InstructorAuditLogsPageComponent],
  imports: [
    CommonModule,
    RouterModule.forChild(routes),
    NgbDatepickerModule,
    FormsModule,
    SessionEditFormModule,
    NgbDropdownModule,
    LoadingSpinnerModule,
    PanelChevronModule,
    SortableTableModule,
  ],
})
export class InstructorAuditLogsPageModule { }
