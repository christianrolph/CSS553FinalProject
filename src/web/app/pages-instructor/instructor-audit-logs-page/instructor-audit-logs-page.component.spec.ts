import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { of } from 'rxjs';

import { CourseService } from '../../../services/course.service';
import { LogService } from '../../../services/log.service';
import { StudentService } from '../../../services/student.service';
import { TimezoneService } from '../../../services/timezone.service';
import {
  Course,
  FeedbackSession,
  FeedbackSessionLog,
  FeedbackSessionPublishStatus,
  FeedbackSessionSubmissionStatus,
  LogType,
  ResponseVisibleSetting,
  SessionVisibleSetting,
  Student,
} from '../../../types/api-output';
import { SortBy } from '../../../types/sort-properties';
import { ColumnData } from '../../components/sortable-table/sortable-table.component';
import { InstructorAuditLogsPageComponent } from './instructor-audit-logs-page.component';
import { InstructorAuditLogsPageModule } from './instructor-audit-logs-page.module';
import Spy = jasmine.Spy;

describe('InstructorAuditLogsPageComponent', () => {
  let component: InstructorAuditLogsPageComponent;
  let fixture: ComponentFixture<InstructorAuditLogsPageComponent>;
  let courseService: CourseService;
  let studentService: StudentService;
  let logService: LogService;
  let timezoneService: TimezoneService;

  const resultColumns: ColumnData[] = [
    { header: 'Time', sortBy: SortBy.LOG_DATE },
    { header: 'Name', sortBy: SortBy.GIVER_NAME },
    { header: 'Activity', sortBy: SortBy.LOG_TYPE },
    { header: 'Email', sortBy: SortBy.RESPONDENT_EMAIL },
    { header: 'Section', sortBy: SortBy.SECTION_NAME },
    { header: 'Team', sortBy: SortBy.TEAM_NAME },
  ];
  const testCourse1: Course = {
    courseId: 'CS9999',
    courseName: 'CS9999',
    timeZone: 'Asia/Singapore',
    creationTimestamp: 0,
    deletionTimestamp: 0,
  };
  const testCourse2: Course = {
    courseId: 'MA1234',
    courseName: 'MA1234',
    timeZone: 'Asia/Singapore',
    creationTimestamp: 0,
    deletionTimestamp: 0,
  };
  const emptyStudent: Student = {
    courseId: '', email: '', name: '', sectionName: '', teamName: '',
  };
  const testStudent: Student = {
    email: 'doejohn@email.com',
    courseId: 'CS9999',
    name: 'Doe John',
    teamName: 'team 1',
    sectionName: 'section 1',
  };
  const testFeedbackSession: FeedbackSession = {
    feedbackSessionName: 'Feedback Session 1',
    courseId: 'CS9999',
    timeZone: 'Asia/Singapore',
    instructions: '',
    submissionStartTimestamp: 0,
    submissionEndTimestamp: 1549095330000,
    gracePeriod: 0,
    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
    publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
    isClosingEmailEnabled: true,
    isPublishedEmailEnabled: true,
    createdAtTimestamp: 0,
  };
  const testLogs1: FeedbackSessionLog = {
    feedbackSessionData: testFeedbackSession,
    feedbackSessionLogEntries: [
      {
        studentData: testStudent,
        feedbackSessionLogType: LogType.FEEDBACK_SESSION_SUBMISSION,
        timestamp: 0,
      },
    ],
  };
  const testLogs2: FeedbackSessionLog = {
    feedbackSessionData: testFeedbackSession,
    feedbackSessionLogEntries: [
      {
        studentData: testStudent,
        feedbackSessionLogType: LogType.FEEDBACK_SESSION_SUBMISSION,
        timestamp: 0,
      },
    ],
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [InstructorAuditLogsPageModule, HttpClientTestingModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorAuditLogsPageComponent);
    courseService = TestBed.inject(CourseService);
    studentService = TestBed.inject(StudentService);
    logService = TestBed.inject(LogService);
    timezoneService = TestBed.inject(TimezoneService);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when page is still loading', () => {
    component.isLoading = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when searching for details in search form', () => {
    component.courses = [testCourse1, testCourse2];
    component.formModel = {
      logsDateFrom: { year: 1997, month: 9, day: 11 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 1998, month: 9, day: 11 },
      logsTimeTo: { hour: 15, minute: 0 },
      courseId: 'CS9999',
      studentEmail: 'doejohn@email.com',
    };
    component.courseToStudents = {
      CS9999: [testStudent],
      MA1234: [],
    };
    component.isLoading = false;
    component.isSearching = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should snap with results of a search', () => {
    component.searchResults = [
      {
        feedbackSessionName: 'Feedback session 1',
        logColumnsData: resultColumns,
        logRowsData: [[
          { value: '15 January 2021' },
          { value: 'Doe John' },
          { value: 'Viewed the submission page' },
          { value: 'doejohn@email.com' },
          { value: 'section 1' },
          { value: 'team 1' },
        ]],
        isTabExpanded: true,
      },
      {
        feedbackSessionName: 'Feedback session 2',
        logColumnsData: resultColumns,
        logRowsData: [],
        isTabExpanded: true,
      },
    ];
    component.isLoading = false;
    component.isSearching = false;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should load all courses and students that instructor has on init', () => {
    const courseSpy: Spy = spyOn(courseService, 'getAllCoursesAsInstructor').and
        .returnValue(of({
          courses: [
            testCourse1, testCourse2,
          ],
        }));
    const studentSpy: Spy = spyOn(studentService, 'getStudentsFromCourse').and
        .returnValue(of({
          students: [
            testStudent,
          ],
        }));

    component.ngOnInit();

    expect(component.isLoading).toBeFalsy();
    expect(courseSpy).toBeCalledWith('active');
    expect(component.courses.length).toEqual(2);
    expect(component.courseToStudents[testCourse1.courseId][0]).toEqual(emptyStudent);
    expect(component.courseToStudents[testCourse1.courseId][1]).toEqual(testStudent);
    expect(studentSpy).toHaveBeenNthCalledWith(1, { courseId: testCourse1.courseId });
    expect(studentSpy).toHaveBeenNthCalledWith(2, { courseId: testCourse2.courseId });
  });

  it('should search for logs when search button is clicked', () => {
    const logSpy: Spy = spyOn(logService, 'searchFeedbackSessionLog').and
        .returnValue(of({ feedbackSessionLogs: [testLogs1, testLogs2] }));
    const timeSpy: Spy = spyOn(timezoneService, 'getResolvedTimestamp').and
        .returnValue(of({ timestamp: 0, message: '' }));

    component.isLoading = false;
    component.isSearching = false;
    component.formModel = {
      logsDateFrom: { year: 2020, month: 12, day: 30 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2020, month: 12, day: 31 },
      logsTimeTo: { hour: 23, minute: 59 },
      courseId: testCourse1.courseId,
      studentEmail: testStudent.email,
    };
    component.courseToStudents = { CS9999: [testStudent] };
    fixture.detectChanges();

    fixture.debugElement.nativeElement.querySelector('#search-button').click();

    expect(timeSpy).toHaveBeenCalledTimes(2);
    expect(logSpy).toHaveBeenCalled();
    expect(logSpy).toHaveBeenCalledWith({
      courseId: testCourse1.courseId, searchFrom: '0', searchUntil: '0', studentEmail: testStudent.email,
    });

    expect(component.searchResults.length).toEqual(2);

    for (let i: number = 0; i < 2; i += 1) {
      expect(component.searchResults[i].isTabExpanded).toBeFalsy();
      expect(component.searchResults[i].logColumnsData).toEqual(resultColumns);
      // Testing that the LogType is converted correctly.
      expect(component.searchResults[i].logRowsData[0][2].value).toEqual('Submitted responses');
    }
  });
});
