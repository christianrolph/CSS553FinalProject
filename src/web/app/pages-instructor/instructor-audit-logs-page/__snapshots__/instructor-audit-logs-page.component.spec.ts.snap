// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorAuditLogsPageComponent should snap when page is still loading 1`] = `
<tm-instructor-audit-logs-page
  LOGS_RETENTION_PERIOD={[Function Number]}
  SortBy={[Function Object]}
  courseService={[Function CourseService]}
  courseToStudents={[Function Object]}
  courses={[Function Array]}
  dateToday={[Function Object]}
  earliestSearchDate={[Function Object]}
  formModel={[Function Object]}
  isLoading={[Function Boolean]}
  isSearching="false"
  logsService={[Function LogService]}
  searchResults={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Student Activity Logs
  </h1><p>
    This page allows you to find when your students have accessed or submitted a particular feedback session for a given course. The earliest date you can search for is 
    <b>
      30 days
    </b>
     before today.
  </p><hr /><tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner><div>
    
  </div>
</tm-instructor-audit-logs-page>
`;

exports[`InstructorAuditLogsPageComponent should snap when searching for details in search form 1`] = `
<tm-instructor-audit-logs-page
  LOGS_RETENTION_PERIOD={[Function Number]}
  SortBy={[Function Object]}
  courseService={[Function CourseService]}
  courseToStudents={[Function Object]}
  courses={[Function Array]}
  dateToday={[Function Object]}
  earliestSearchDate={[Function Object]}
  formModel={[Function Object]}
  isLoading="false"
  isSearching={[Function Boolean]}
  logsService={[Function LogService]}
  searchResults={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Student Activity Logs
  </h1><p>
    This page allows you to find when your students have accessed or submitted a particular feedback session for a given course. The earliest date you can search for is 
    <b>
      30 days
    </b>
     before today.
  </p><hr /><div
    class="card bg-form no-border"
  >
    <div
      class="card-body"
    >
      <div
        class="row form-group"
      >
        <div
          class="col-4"
        >
          <div
            class="col-12"
          >
            <label
              class="font-weight-bold"
              for="course-id-dropdown"
            >
              Course ID
            </label>
          </div>
          <div
            class="col-12"
          >
            <select
              class="form-control col-12 ng-untouched ng-pristine ng-valid"
              id="course-id-dropdown"
            >
              <option
                disabled=""
                hidden=""
                selected=""
                value=""
              >
                Select Course ID
              </option>
              
              <option>
                 CS9999 
              </option>
              <option>
                 MA1234 
              </option>
            </select>
          </div>
        </div>
        <div
          class="col-8"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-from-datepicker"
            >
              Search period from
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-from-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row form-group"
      >
        <div
          class="col-4"
        >
          <div
            class="col-12"
          >
            <label
              class="font-weight-bold"
              for="student-name-dropdown"
            >
              Student Name (Optional)
            </label>
          </div>
          <div
            class="col-12"
          >
            <select
              class="form-control col-12 ng-untouched ng-pristine ng-valid"
              id="student-name-dropdown"
            >
              
              <option
                value="doejohn@email.com"
              >
                 Doe John &lt;doejohn@email.com&gt; 
              </option>
            </select>
          </div>
        </div>
        <div
          class="col-8"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-to-datepicker"
            >
              Search period until
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-to-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
            <div
              class="col-md-2"
            >
              <button
                class="float-right btn btn-primary"
                disabled=""
                id="search-button"
              >
                Search
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div><tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner>
</tm-instructor-audit-logs-page>
`;

exports[`InstructorAuditLogsPageComponent should snap with default fields 1`] = `
<tm-instructor-audit-logs-page
  LOGS_RETENTION_PERIOD={[Function Number]}
  SortBy={[Function Object]}
  courseService={[Function CourseService]}
  courseToStudents={[Function Object]}
  courses={[Function Array]}
  dateToday={[Function Object]}
  earliestSearchDate={[Function Object]}
  formModel={[Function Object]}
  isLoading={[Function Boolean]}
  isSearching="false"
  logsService={[Function LogService]}
  searchResults={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Student Activity Logs
  </h1><p>
    This page allows you to find when your students have accessed or submitted a particular feedback session for a given course. The earliest date you can search for is 
    <b>
      30 days
    </b>
     before today.
  </p><hr /><tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner><div>
    
  </div>
</tm-instructor-audit-logs-page>
`;

exports[`InstructorAuditLogsPageComponent should snap with results of a search 1`] = `
<tm-instructor-audit-logs-page
  LOGS_RETENTION_PERIOD={[Function Number]}
  SortBy={[Function Object]}
  courseService={[Function CourseService]}
  courseToStudents={[Function Object]}
  courses={[Function Array]}
  dateToday={[Function Object]}
  earliestSearchDate={[Function Object]}
  formModel={[Function Object]}
  isLoading="false"
  isSearching="false"
  logsService={[Function LogService]}
  searchResults={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Student Activity Logs
  </h1><p>
    This page allows you to find when your students have accessed or submitted a particular feedback session for a given course. The earliest date you can search for is 
    <b>
      30 days
    </b>
     before today.
  </p><hr /><div
    class="card bg-form no-border"
  >
    <div
      class="card-body"
    >
      <div
        class="row form-group"
      >
        <div
          class="col-4"
        >
          <div
            class="col-12"
          >
            <label
              class="font-weight-bold"
              for="course-id-dropdown"
            >
              Course ID
            </label>
          </div>
          <div
            class="col-12"
          >
            <select
              class="form-control col-12 ng-untouched ng-pristine ng-valid"
              id="course-id-dropdown"
            >
              <option
                disabled=""
                hidden=""
                selected=""
                value=""
              >
                Select Course ID
              </option>
              
            </select>
          </div>
        </div>
        <div
          class="col-8"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-from-datepicker"
            >
              Search period from
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-from-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row form-group"
      >
        <div
          class="col-4"
        >
          <div
            class="col-12"
          >
            <label
              class="font-weight-bold"
              for="student-name-dropdown"
            >
              Student Name (Optional)
            </label>
          </div>
          <div
            class="col-12"
          >
            <select
              class="form-control col-12 ng-untouched ng-pristine ng-valid"
              id="student-name-dropdown"
            >
              
            </select>
          </div>
        </div>
        <div
          class="col-8"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-to-datepicker"
            >
              Search period until
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-to-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
            <div
              class="col-md-2"
            >
              <button
                class="float-right btn btn-primary"
                disabled=""
                id="search-button"
              >
                Search
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div><div>
    
    <div>
      <div
        class="card mb-3 alert-success-border"
      >
        <div
          class="card-header alert-success cursor-pointer"
        >
           Feedback session 1 
          <div
            class="card-header-btn-toolbar"
          >
            <tm-panel-chevron>
              <div
                class="chevron"
              >
                
                
                <i
                  class="fas fa-chevron-up"
                />
              </div>
            </tm-panel-chevron>
          </div>
        </div>
        
        <div>
          
          <div
            class="card-body"
          >
            <tm-sortable-table>
              <table
                class="table table-bordered table-striped margin-bottom-20px"
              >
                <thead>
                  <tr>
                    
                    <th
                      class="sortable-header"
                    >
                      <span
                        container="body"
                      >
                        Time
                      </span>
                      
                      <span
                        class="fa-stack"
                      >
                        <i
                          class="fas fa-sort"
                        />
                        
                        
                      </span>
                    </th>
                    <th
                      class="sortable-header"
                    >
                      <span
                        container="body"
                      >
                        Name
                      </span>
                      
                      <span
                        class="fa-stack"
                      >
                        <i
                          class="fas fa-sort"
                        />
                        
                        
                        <i
                          class="fas fa-sort-up"
                        />
                      </span>
                    </th>
                    <th
                      class="sortable-header"
                    >
                      <span
                        container="body"
                      >
                        Activity
                      </span>
                      
                      <span
                        class="fa-stack"
                      >
                        <i
                          class="fas fa-sort"
                        />
                        
                        
                      </span>
                    </th>
                    <th
                      class="sortable-header"
                    >
                      <span
                        container="body"
                      >
                        Email
                      </span>
                      
                      <span
                        class="fa-stack"
                      >
                        <i
                          class="fas fa-sort"
                        />
                        
                        
                      </span>
                    </th>
                    <th
                      class="sortable-header"
                    >
                      <span
                        container="body"
                      >
                        Section
                      </span>
                      
                      <span
                        class="fa-stack"
                      >
                        <i
                          class="fas fa-sort"
                        />
                        
                        
                      </span>
                    </th>
                    <th
                      class="sortable-header"
                    >
                      <span
                        container="body"
                      >
                        Team
                      </span>
                      
                      <span
                        class="fa-stack"
                      >
                        <i
                          class="fas fa-sort"
                        />
                        
                        
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  
                  <tr>
                    
                    <td>
                      
                      
                      
                      
                      15 January 2021
                    </td>
                    <td>
                      
                      
                      
                      
                      Doe John
                    </td>
                    <td>
                      
                      
                      
                      
                      Viewed the submission page
                    </td>
                    <td>
                      
                      
                      
                      
                      doejohn@email.com
                    </td>
                    <td>
                      
                      
                      
                      
                      section 1
                    </td>
                    <td>
                      
                      
                      
                      
                      team 1
                    </td>
                  </tr>
                </tbody>
              </table>
            </tm-sortable-table>
          </div>
          
        </div>
      </div>
    </div>
    <div>
      <div
        class="card mb-3 alert-success-border"
      >
        <div
          class="card-header alert-success cursor-pointer"
        >
           Feedback session 2 
          <div
            class="card-header-btn-toolbar"
          >
            <tm-panel-chevron>
              <div
                class="chevron"
              >
                
                
                <i
                  class="fas fa-chevron-up"
                />
              </div>
            </tm-panel-chevron>
          </div>
        </div>
        
        <div>
          
          <div
            class="no-activity-body"
          >
            <i>
              No activity for this feedback session in selected search period
            </i>
          </div>
          
        </div>
      </div>
    </div>
  </div>
</tm-instructor-audit-logs-page>
`;
